<a href="https://colab.research.google.com/github/GioKhoma/FitTech_Analysis/blob/main/FitTech_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**1.**
**შესავალი:**

FitTech არის სავარჯიშო აპი, რომელიც აკონტროლებს ვარჯიშის პროცესს, იგი უკვე 3 წელია ფუნქციონირებს და მიზნად ისახავს მომხმარებლის ჩართულობისა და ფიტნეს მიღწევების გაზრდას. ჩვენს ხელთაა მომხმარებლის აქტივობის გასული წლის მონაცემები, ჩვენი მიზანია გავიგოთ, თუ რამდენად კარგად იყენებენ მომხმარებლები ამ აპს და რა გავლენას ახდენს მისი მოხმარება ვარჯიშზე. მოცემული ანალიზის დანიშნულებაა, რომ გავიგოთ აპლიკაციის ეფექტურობა და ის მთავარი კომპონენტები, რომლებიც მომავალში აპლიკაციის განითარებას შეუწყობენ ხელს.


**2. ზოგადი შედეგები**

ქვემოთ მოცემული კოდის საშუალებით გავიგებთ, აქტიური მომხმარებლების საერთო რაოდენობასა და ჩატარებული ვარჯიშების სესიების რაოდენობას.

In [ ]:
import pandas as pd

df = pd.read_csv('/content/sample_data/fittech_data.csv')

total_active_users = df['მომხმარებლის ID'].nunique()
total_workouts = len(df)

print(f"აქტიური მომხმარებლების საერთო რაოდენობა: {total_active_users:,}")
print(f"ვარჯიშის სესიების საერთო რაოდენობა: {total_workouts:,}")

აქტიური მომხმარებლების საერთო რაოდენობა: 10,000
ვარჯიშის სესიების საერთო რაოდენობა: 477,481


შედეგად მივიღეთ, რომ აქტიური მომხმარებლების რაოდენობა უდრი 10000 -ს და სესიების საერთო რაოდენობაა 477481.

**3. ვარჯიშის ტიპების ეფექტურობა**

განვიხილოთ, თუ როგორ შეუწყო ხელი თითოეულმა ვარჯიშის ტიპმა მომხმარებელთა ჩართულობას:

In [ ]:
workout_performance =  df.groupby('ვარჯიშის ტიპი').agg(
    საშუალო_ხანგრძლივობა = ('ხანგრძლივობა', 'mean'),
    ჯამური_ხანგრძლივობა = ('ხანგრძლივობა', 'sum'),
    ჯამური_კალორიები = ('დამწვარი კალორიები', 'sum'),
    სესიები = ('ვარჯიშის ტიპი', 'count')
)

workout_performance.columns = ['საშუალო ხანგრძლივობა', 'ჯამური ხანგრძლივობა', 'ჯამური კალორიები', 'სესიები']

workout_performance = workout_performance.sort_values(by='სესიები', ascending=False)

print(workout_performance)

               საშუალო ხანგრძლივობა  ჯამური ხანგრძლივობა  ჯამური კალორიები  \
ვარჯიშის ტიპი                                                                
სირბილი                   37.497889         6.271859e+06      4.704373e+07   
იოგა                      67.510458         6.469055e+06      3.234402e+07   
ველოსიპედი                37.444090         3.572990e+06      1.787051e+07   
ცურვა                     37.469173         2.663908e+06      1.330985e+07   
ძალოსნობა                 37.521945         1.796588e+06      1.077242e+07   

               სესიები  
ვარჯიშის ტიპი           
სირბილი         167259  
იოგა             95823  
ველოსიპედი       95422  
ცურვა            71096  
ძალოსნობა        47881  


**მიგნებები:**

ანალიზმა დაგვანახა, რომ მომხმარებლები ყველაზე დიდ დროს უთმობენ იოგას, ხოლო ყველაზე მეტი კალორია დაწვეს მორბენლებმა.

**რეკომენდაციები:**

იმისთვის, თუ მომხმარებლეს სურს მეტი კალორიის დაწვა, რეკომენდაციის სახით ვურჩევთ, რომ მისთვის კარგი აქტივობა იქნებოდა სირბილი.

**4. მომხმარებელთა კმაყოფილება**

ახლა ვნახოთ, როგორ აფასებენ მომხმარებლები სხვადასხვა ტიპის ვარჯიშებს:


In [ ]:
workout_ratings = df.groupby('ვარჯიშის ტიპი')['მომხმარებლის შეფასება'].mean()

workout_ratings = workout_ratings.sort_values(ascending=False)

print(workout_ratings)

ვარჯიშის ტიპი
ველოსიპედი    4.001352
სირბილი       3.505079
ცურვა         3.503530
იოგა          3.501863
ძალოსნობა     2.499488
Name: მომხმარებლის შეფასება, dtype: float64


**დაკვირვებები:**

მოცემული ინფორმაციის საფუძველზე, ვასკვნით, რომ ყველაზე მაღალი რეიტინგი აქვს ველოსპორტს, ხოლო ყველაზე დაბალი რეიტინგი ძალოსნობას აქვს.

**შეხედულება:**

მოცემული შედეგები მეტყველებს სპორტის სირთულეებზე, რადგან ველომოყვარული ადამიანი ბევრად მარტივად იკმაყოფილებს თავის სურვილებს, ველოსიპედით სეირნობისას, ვიდრე ძალოსანი.

**5. ვარჯიშის ხანგრძლივობისა და კალორიების წვის კორელაცია**

განვიხილოთ, არის თუ არა კავშირი ვარჯიშის ხანგრძლივობასა და დამწვარ კალორიებს შორის:

In [ ]:
correlation = df['ხანგრძლივობა'].corr(df['დამწვარი კალორიები'])

print(f"კორელაცია ხანგრძლივობასა და დამწვარ კალორიებს შორის: {correlation:.2f}")

კორელაცია ხანგრძლივობასა და დამწვარ კალორიებს შორის: 0.86


**ანალიზი:**

კორელაცია ხანგრძლივობასა და დამწვარ კალორიებს შორის უდრი 0.86. მოცემული შედეგი მეტყველებს იმაზე რომ უფრო ხანგრძლივი ვარჯიში დაკავშირებულია მაღალ კალორიულ ხარჯებთან.

**სტრატეგია:**

ამ კორელაციის გათვალისწინებით აპლიკაციამ ყურადღება უნდა გაამახვილოს მომხმარებელთა წახალისებაზე, რომ გააუმჯობესონ ხანგრძლივობა.

**6. ყოველთვიური ჩართულობის ტენდენციები**

განვიხილოთ, როგორ მერყეობს მომხმარებელთა ჩართულობა წლის განმავლობაში:

In [ ]:
df['თარიღი'] = pd.to_datetime(df['თარიღი'])

df['თვე'] = df['თარიღი'].dt.to_period('M')

monthly_engagement = df.groupby('თვე').agg(
    სესია=('ვარჯიშის ტიპი', 'count'),
    საშუალო_ხანგრძლივობა=('ხანგრძლივობა', 'mean'),
    საშუალო_დამწვარი_კალორიები=('დამწვარი კალორიები', 'mean')
)

monthly_engagement.columns = ['სესიები', 'საშუალო ხანგრძლივობა', 'საშუალო კალორიები']
monthly_engagement = monthly_engagement.sort_values(by='სესიები', ascending=False)

print(monthly_engagement)

best_month = monthly_engagement.index[0]
worst_month = monthly_engagement.index[-1]

print(f"\nსაუკეთესო თვე: {best_month}")
print(f"ყველაზე ცუდი თვე: {worst_month}")

         სესიები  საშუალო ხანგრძლივობა  საშუალო კალორიები
თვე                                                      
2023-01    42562             43.636475         254.700382
2023-05    42554             43.475953         254.345377
2023-10    42552             43.443936         253.868628
2023-03    42470             43.435890         253.659509
2023-07    42421             43.389336         253.156022
2023-09    41361             43.527582         254.030062
2023-11    41134             43.570786         254.487311
2023-06    40948             43.572326         254.867099
2023-04    40839             43.505032         254.122083
2023-08    39607             43.471028         253.754549
2023-02    38347             43.541480         254.459107
2023-12    22686             43.558701         254.089520

საუკეთესო თვე: 2023-01
ყველაზე ცუდი თვე: 2023-12


**დაკვირვებები:**

  ანალიზის შედეგად ვასკვნით, რომ იანვარში ყველაზე მეტი ადამიანი იწყებს ვარჯიშს,
  ხოლო წლის მიწურულს, დეკემბერში, ყველაზე ცოტა სესია ფიქსირდება.

**შეხედულება:**

  ვფიქრობ, მოცემული ანალიზის გამომწვევი მიზანია ახალი წლის საწყისი, რადგან ამ დროს ადამიანები ბევრად მოტივირებულები არიან, ამასთან ერთად მრავალი დღესასწაულის შედეგად მომატებულ კილოგრამების დაკლებას ცდილობენ.

**სტრატეგია:**

წამახალისებელი სტრატეგია წლის საწყის თვეებში შეიძლება იყოს აბონიმენტის ფასდაკლება, საჩუქრები და ა.შ.

**7. მომხმარებელთა თანმიმდევრულობის ანალიზი**

გავაანალიზოთ მომხმარებელთა ვარჯიშის სიხშირეთა განაწილება:

In [ ]:
user_frequency = df.groupby('მომხმარებლის ID')['თარიღი'].count()
frequency_stats = user_frequency.describe()
frequency_quantiles = user_frequency.quantile([0.25, 0.5, 0.75])

print("ვარჯიშის სიხშირის სტატისტიკა:")
print(frequency_stats)
print("\nვარჯიშის სიხშირის კვანტილები:")
print(frequency_quantiles)

ვარჯიშის სიხშირის სტატისტიკა:
count    10000.000000
mean        47.748100
std         15.785087
min         12.000000
25%         36.000000
50%         45.000000
75%         57.000000
max        127.000000
Name: თარიღი, dtype: float64

ვარჯიშის სიხშირის კვანტილები:
0.25    36.0
0.50    45.0
0.75    57.0
Name: თარიღი, dtype: float64


**ანალიზი:**

ტრენინგის საშუალო სიხშირე წელიწადში საშუალოდ 47.75 სესიაზე მიუთითებს, ხოლო სტანდარტული გადახრა უდრის 15.7-ს.

მომხმარებელთა 25% ატარებს 36 სესიან ან ნაკლებს წეილიწადში, ხოლო 75% აქვს 57 ან ნაკლები სესია წელიწადში.

მედიანური სიხშირე არის 45 სესიის ტოლი, რაც გვაჩვენებს, რომ მომხმარებელთა ნახევარი ვარჯიშობს ამ ოდენობაზე ნაკლებს ან ტოლს.


**რეკომენდაცია:**

ნაკლებად აქტიური მომხმარებლების ჩართულობისთვის უნდა მოქმედებდეს წამახალისებელი ქმედებები, მაგალითად, თუ მომხმარებელს არ უვარჯიშია 3 თვე, აპლიკაციის სახით შეიძლება მიუვიდეს შეტყობინება ფასდაკლების შესახებ, წამახალისებელი ასევე შეიძლება იყოს სხვა ნებისმიერი საჩუქარი, რომელსაც მომხმარებელი ვარჯიშის დაწყებისას მიიღებს.



**9. დასკვნა**

მოცემული ანალიზის შედეგად შეგვიძლია დავასკვნათ, რომ 2023 წელი FitTech-ისთვის საკმაოდ წარმატებული გამოდგა.

სპორტულ სახეობებში ლიდერობდა სირბილი, რომელზეც ყველაზე მეტი სესია დაფიქსირდა, მისი პიკი კი წლის დასაწყისში იყო.

რაც შეეხება მომხმარებელთა ჩართულობის გაზრდას, როგორც ზევით ვახსენეთ კარგი იქნება სხვადასხვა წამახალისებელი კამპანიების წამოწყება, რომელსაც დადებითი შედეგი ექნება როგორც აპლიკაციისათვის, ასევე მომხმარებლებისთვის.

ანალიზის შედეგად გავიგეთ თუ რომელი სპორტის სახეობაა ყველაზე მოთხოვნადი მომხმარებლებში და რომელი არის უკეთესი კალორიების დაწვის თვალსაზრისით, ამასთან ერთად გავარკვიეთ რომელ თვეებშია არიან ადამიანები სპორტულად ყველაზე აქტიურები.

რეკომენდაციის სახით შეგვიძლია ვთქვათ, რომ უკეთესი იქნება წამახალისებელი კამპანიების წამოწყება, მომხმარებელთა რეიტინგის გაზიარება, რათა მომხმარებლებს ქონდეთ მეტი მოტივაცია.

ანალიზის შედეგან უამრავი რამის დასკვნა შეგვიძლია, რომლების გათვალისწინების შედეგან აპლიკაცია ბევრად წარმატებული გახდება და მეტი მომხმარებელი ისარგებლებს მისი სერვისით.